### PR TITLE
Fix saving null values for module extra data

### PR DESCRIPTION
**Pull request by jonasdekeukelaere**
## Type
- Non critical bugfix

## Pull request description
Denies duplicate modules extras.

### CODE REVIEW

File: src/Backend/Core/Engine/Model.php [hunk 1]
```diff
@@ -708,7 +708,7 @@ public static function insertExtra(
                 'type' => $type,
                 'label' => $label ?? $module, // if label is empty, fallback to module
                 'action' => $action ?? $module, // if action is empty, fallback to module
-                'data' => serialize($data),
+                'data' => $data ===  null ? null : serialize($data),
```
Comment (carakas) [inline src/Backend/Core/Engine/Model.php]: is this also checked when unserialising ?

Comment (jonasdekeukelaere) [inline src/Backend/Core/Engine/Model.php]: Fixed in 52a15bead

File: src/Backend/Core/Engine/Model.php [hunk 2]
```diff
@@ -389,7 +389,7 @@ public static function getExtrasForData(string $module, string $key, string $val
             array_filter(
                 $moduleExtras,
                 function (array $data) use ($key, $value) {
-                    $data = unserialize($data);
+                    $data = $data === null ? [] : unserialize($data);
```
Comment (carakas) [inline src/Backend/Core/Engine/Model.php]: wouldn't it make more sense to return null if the data is null ?

Comment (jonasdekeukelaere) [inline src/Backend/Core/Engine/Model.php]: Didn't change anything to the logic, as the return value should be a bool ` return isset($data[$key]) && $data[$key] === $value;` (filter method)

Comment (carakas) [inline src/Backend/Core/Engine/Model.php]: ah k, sorry, missed that

**Pull request merged**